---
 lib/binsrch.c |    4 	2 +	2 -	0 !
 src/wn.c      |    5 	2 +	3 -	0 !
 2 files changed, 4 insertions(+), 5 deletions(-)

Index: WordNet-3.0/lib/binsrch.c
===================================================================
--- WordNet-3.0.orig/lib/binsrch.c	2009-08-24 05:37:29.000000000 -0400
+++ WordNet-3.0/lib/binsrch.c	2009-08-24 05:49:27.000000000 -0400
@@ -193,7 +193,7 @@ char *replace_line(char *new_line, char 
     copyfile(fp, tfp);
     if (fseek(fp, offset, 0) == -1)
 	return(NULL);		/* could not seek to offset */
-    fprintf(fp, new_line);	/* write line */
+    fprintf(fp, "%s", new_line);	/* write line */
     rewind(tfp);
     copyfile(tfp, fp);
 
@@ -220,7 +220,7 @@ char *insert_line(char *new_line, char *
     copyfile(fp, tfp);
     if (fseek(fp, offset, 0) == -1)
 	return(NULL);		/* could not seek to offset */
-    fprintf(fp, new_line);	/* write line */
+    fprintf(fp, "%s", new_line);	/* write line */
     rewind(tfp);
     copyfile(tfp, fp);
 
Index: WordNet-3.0/src/wn.c
===================================================================
--- WordNet-3.0.orig/src/wn.c	2009-08-24 05:37:29.000000000 -0400
+++ WordNet-3.0/src/wn.c	2009-08-24 05:53:23.000000000 -0400
@@ -283,8 +283,7 @@ static void printsearches(char *word, in
 	    printf("\t");
 	    printf(searchstr[j].template,
 		   partchars[dbase], partchars[dbase]);
-	    printf(searchstr[j].helpstr);
-	    printf("\n");
+	    printf("%s\n", searchstr[j].helpstr);
 	}
 }
 
@@ -344,7 +343,7 @@ static int getoptidx(char *searchtype)
 
 static int error_message(char *msg)
 {
-    fprintf(stderr, msg);
+    fprintf(stderr, "%s\n", msg);
     return(0);
 }
 
